react-components/src/context/UserContext.js

```diff
@@ -217,12 +218,7 @@ const UserContextProvider = props => {
     const [userCookie, setUserCookie] = useCookieValue('cif.userToken');
     const [, setCartCookie] = useCookieValue('cif.cart');
     const isSignedIn = () => !!userCookie;
-    const accountContainerQuerySelector =
-        (props.config && props.config.accountContainerQuerySelector) || '.header__accountTrigger #miniaccount';
-    const addressBookContainerQuerySelector =
-        (props.config && props.config.addressBookContainerQuerySelector) || '#addressbook';
-    const addressBookPath =
-        (props.config && props.config.addressBookPath) || '/content/venia/us/en/my-account/address-book.html';
+    const { pagePaths } = useConfigContext();
```

Comment:
A context depending on other context is a bad idea IMO. I understand why it needs to be done here, but we need to refactor the UserContext anyway.

Reply:
Actually it's a common pattern. But I agree. Just look at all the tests I had to adapt :D 

We would basically need to move https://github.com/adobe/aem-core-cif-components/blob/c7ff96a7a879d518d4c79b3b7795966b786b8710/react-components/src/context/UserContext.js#L316-L318 somewhere else. Maybe we can remove it from the context and have every component that uses it implement it by taking the path from the new `ConfigContext`.

Reply:
I think that's a better approach. UserContext holds too much information as it is.

Reply:
Okay, I refactored both `UserContext` and `NavigationContext` to not depend on `ConfigContext` anymore :)